src/Solvers/conjugate_gradient_poisson_solver.jl

```diff
@@ -34,33 +34,47 @@ end
     @inbounds ∇²ϕ[i, j, k] = ∇²ᶜᶜᶜ(i, j, k, grid, ϕ)
 end
 
-function compute_laplacian!(∇²ϕ, ϕ)
+struct RegularizedLaplacian{D}
+    δ :: D
+end
+
+function (L::RegularizedLaplacian)(Lϕ, ϕ)
     grid = ϕ.grid
     arch = architecture(grid)
     fill_halo_regions!(ϕ)
-    launch!(arch, grid, :xyz, laplacian!, ∇²ϕ, grid, ϕ)
+    launch!(arch, grid, :xyz, laplacian!, Lϕ, grid, ϕ)
+
+    if !isnothing(L.δ)
+        # Add regularization
+        ϕ̄ = mean(ϕ)
+        parent(Lϕ) .+= L.δ * ϕ̄
```

Comment:
Do we need to mask `Lϕ` after this operation? It seems to that the masked area also gets a non-zero value after this line.

Reply:
Oh, good catch. Actually, while `dot` and `norm` _should_ ignore immersed regions, I'm actually not sure they do right now so this may matter. I will test it.